h2o-core/src/main/java/water/parser/CsvParser.java

```diff
@@ -18,14 +18,19 @@
   private static final int NO_HEADER = ParseSetup.NO_HEADER;
   private static final int GUESS_HEADER = ParseSetup.GUESS_HEADER;
   private static final int HAS_HEADER = ParseSetup.HAS_HEADER;
-  private static final byte[] NON_DATA_LINE_MARKERS = {'#'};
+  private static final byte[] NON_DATA_LINE_MARKERS_DEFAULT = {'#'};
 
-  CsvParser( ParseSetup ps, Key jobKey ) { super(ps, jobKey); }
+  CsvParser( ParseSetup ps, Key jobKey ) {
+    super(ps, jobKey);
+    if(_setup._nonDataLineMarkers == null) _setup._nonDataLineMarkers = nonDataLineMarkersDefault();
+  }
 
-  protected byte[] nonDataLineMarkers() {
-    return NON_DATA_LINE_MARKERS;
+  protected byte[] nonDataLineMarkersDefault() {
```

Comment:
this method is useless I think

Reply:
It is not. This behavior is overridden in `ARFFParser`.

Reply:
Unfortunately, it is not. The behavior is overridden in `ARFFParser`, which calls CSVParser's constructor in it's own constructor. Each class has different defaults to provide.